### PR TITLE
docs: add profile/README.md

### LINKED ISSUE #1: Add an organization README.md
As discussed with the maintainers, the purpose of this issue is to create a README.md for the org which provides context about how development and building for OSSM 3.x is done, perhaps some historical context, and overall just some pointers to anyone who comes to the org to understand it's function.

### PR DESCRIPTION
**Pull request by shaneutt**
Adds a README.md for the Github org to resolve #1.

This initial README is mostly topical as much of the development for OSSM actually happens in upstream. So mostly what you'll find here are some pointers and notes to help a new OSSM developer get to the right places a bit faster.

### CODE REVIEW

File: profile/README.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+> Container Platform Documentation][ocp-docs] for your version instead.
+
+[Istio]:https://github.com/istio
+[ossm-istio]:https://github.com/openshift-service-mesh/istio
+[ocp-docs]:https://docs.redhat.com/en/documentation/openshift_container_platform/
+
+## Development
+
+The vast majority of development of OpenShift Service Mesh (OSSM) happens in
+[upstream Istio], the exception being some areas related to [Envoy Proxy] or
```
Comment (npinaeva) [inline profile/README.md:35]: may be nice to add a short explanation of how istio relates to envoy before going into more details

**Review by npinaeva — COMMENTED** [review]
Really nice docs!

File: profile/README.md (hunk 1 above)
Comment (shaneutt) [reply profile/README.md:35]: Sounds good, I did mention this as the first point in the `## About` section, what do you think should I change that section? Or do I need to just expand further here?

Comment (npinaeva) [reply profile/README.md:35]: I think just saying "Istio uses envoy as ..." in the `About` section or here in the beginning

Comment (shaneutt) [reply profile/README.md:35]: Sounds good, lmkwyt:

```suggestion
[upstream Istio], the exception being some areas related to [Envoy Proxy] (which Istio uses) or
```

Comment (npinaeva) [reply profile/README.md]: maybe add this link https://istio.io/latest/docs/ops/deployment/architecture/ or copy a sentence from there saying "The Istio data plane is composed of a set of intelligent proxies ([Envoy](https://www.envoyproxy.io/)) deployed as sidecars." ?

Comment (shaneutt) [reply profile/README.md]: Sounds good, I pushed e2b4e2a453d7c5d98ab48b28492a0d537a4dbbf2 with changes, LMKWYT?

Comment (npinaeva) [reply profile/README.md]: perrrrfect!

File: profile/README.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+[Istio Ecosystem]:https://github.com/istio-ecosystem/
+[Kubernetes operator]:https://www.redhat.com/en/topics/containers/what-is-a-kubernetes-operator
+[sail]:https://github.com/istio-ecosystem/sail-operator
+[ossm-sail]:https://github.com/openshift-service-mesh/sail-operator
+
+### openshift-service-mesh/proxy
+
+Upstream Istio includes the [istio/proxy][proxy] repository which provides
+Istio-specific options over [Envoy Proxy]. For OpenShift we wrap this repository
+as well in [openshift-service-mesh/proxy][ossm-proxy]. Most of the differences
```
Comment (FilipB) [inline profile/README.md:141]: Broken link

**Review by FilipB — CHANGES_REQUESTED** [review]

File: profile/README.md (hunk 2 above)
Comment (shaneutt) [reply profile/README.md:143]: Thanks! fixed: c030fafd49195a7adce4b6ff18663d2bdc393614

File: profile/README.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+> users**. If you're a user looking to install and use OSSM, see the [OpenShift
+> Container Platform Documentation][ocp-docs] for your version instead.
+
+[Istio]:https://github.com/istio
+[ossm-istio]:https://github.com/openshift-service-mesh/istio
+[ocp-docs]:https://docs.redhat.com/en/documentation/openshift_container_platform/
+
+## Development
+
+The vast majority of development of OpenShift Service Mesh (OSSM) happens in
```
[the remark this answers is not in the document]
Comment (shaneutt) [reply profile/README.md:34]: Done in c030fafd49195a7adce4b6ff18663d2bdc393614, LMKWYT

**Review by FilipB — APPROVED** [review]
Thank you

File: profile/README.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+[upstream Istio] (importantly this includes the [Sail Operator], which is at the
+center of how we deploy and manage the lifecycle of Istio), the most notable
+exceptions being some areas related to [Envoy Proxy] (which are deployed as
+sidecars to [Pods] within the mesh to control and shape traffic (See the [Istio
+Architecture] documentation for more details)) or other building and testing
```
Comment (gcs278) [inline profile/README.md:39]: *nit*
```suggestion
Architecture] documentation for more details) or other building and testing
```

**Review by gcs278 — COMMENTED** [review]
Very helpful doc. I have nothing of value to add.

**Review by dgn — APPROVED** [review]
Thank you, this is great

**Pull request merged**